src/extension.ts

```diff
@@ -10,10 +12,39 @@
 import * as vscode from 'vscode';
 import * as path from 'path';
 import * as fs from 'fs';
+import { isOpenRaInstallationFolder, isOpenRaRepositoryFolder } from './folderUtils';
 
 export async function activate(context: vscode.ExtensionContext) {
 
-	const apiPath = path.resolve(context.extensionPath, "api");
+	// Only activate if the current workspace is an OpenRA-related one.
+	// Using ORAIDE's language server's checks for "is this an OpenRA folder / OpenRA ModSDK folder / etc." might be a bit much,
+	// so using a simpler version of the check, inspired by the ORAIDE VSCode extension.
+	const modFiles = await vscode.workspace.findFiles('**/mod.yaml');
+	if (!modFiles.some(f => fs.readFileSync(f.fsPath, 'utf8').includes('\t'))) {
+		return;
+	}
+
+	let openRaVersion = "dev";
+	
+	// Try to guess what version of OpenRA this is.
+	// If it is an OpenRA repository or an install folder, there will be a "VERSION" file.
+	// If it is a ModSDK repository, there won't be a "VERSION" file (probably). Just go with dev version then.
```

Comment:
You can parse https://github.com/OpenRA/OpenRAModSDK/blob/master/mod.config#L12 or `engine/VERSION`.

Reply:
We can't guarantee that there will be an engine available at the time the extension activates, but reading `mod.config` should work. 👍